modules/tinymce/src/plugins/lists/main/ts/core/Util.ts

```diff
@@ -1,3 +1,27 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
 import { HTMLElement } from '@ephox/dom-globals';
+import { Arr } from '@ephox/katamari';
+import Editor from 'tinymce/core/api/Editor';
+import * as NodeType from './NodeType';
+
+export const isCustomList = (list: HTMLElement) => /\btox\-/.test(list.className);
+
+export const listState = (editor: Editor, listName: string, activate: (active: boolean) => void) =>
+  () => {
+    const nodeChangeHandler = (e) => {
+      const inList = Arr.findUntil(e.parents, NodeType.isListNode, NodeType.isTableCellNode)
+        .map((list: HTMLElement) => list.nodeName === listName && !isCustomList(list))
+        .getOr(false);
```

Comment:
Looks much better!

Note that instead of `x.map().getOr(false);` I usually do `x.filter().isSome()` to avoid the extra option allocation but it's not a huge difference 🤷‍♂️

Reply:
Cool, I'll change it over as I did have something similar at one stage, but generally `isSome` seemed to have been frowned upon for code style so went with this :shrug:

Reply:
I used to frown on it, but realistically you need a boolean here so anything else is just extra noise